core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -59,6 +60,19 @@
     public static final String INDEX_TRANSLOG_DURABILITY = "index.translog.durability";
     public static final String INDEX_REFRESH_INTERVAL = "index.refresh_interval";
     public static final TimeValue DEFAULT_REFRESH_INTERVAL = new TimeValue(1, TimeUnit.SECONDS);
+    public static final String INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE = "index.translog.flush_threshold_size";
+    public static final TimeValue DEFAULT_GC_DELETES = TimeValue.timeValueSeconds(60);
+
+    /**
+     * Index setting to control if a flush is executed before engine is closed
+     * This setting is realtime updateable.
+     */
+    public static final String INDEX_FLUSH_ON_CLOSE = "index.flush_on_close";
```

Comment:
we use this setting mainly in tests but I might be able to work around that. it totally doesn't need to be updateable. I will do this in a followup